core/controllers/domain_objects_validator.py

```diff
@@ -86,17 +99,50 @@ def validate_change_dict_for_blog_post(change_dict):
             raise Exception(
                 'Invalid tags provided. Tags not in default tags list.')
 
+    return change_dict
+
 
 def validate_collection_change(obj):
     # type: (Dict[String, Any]) -> None
     """Validates collection change.
 
     Args:
         obj: dict. Data that needs to be validated.
+
+    Returns:
+        CollectionChange. Returns CollectionChange object after validation.
     """
     # No explicit call to validate_dict method is necessary, because
     # CollectionChange calls validate method while initialization.
-    collection_domain.CollectionChange(obj)
+    return collection_domain.CollectionChange(obj)
+
+
+def validate_task_entry_for_improvements(task_entry_dict):
+    # type: (Dict[String, Any]) -> None
+    """Validates collection change.
+
+    Args:
+        task_entry_dict: dict. Data that needs to be validated.
+
+    Returns:
+        dict(str, any). Returns task_entry_dict after validation.
+    """
+    # There is no validate method in domain layer.
+    entity_version = task_entry_dict.get('entity_version', None)
+    if entity_version is None:
+        raise base.BaseHandler.InvalidInputException(
+            'No entity_version provided')
+    task_type = task_entry_dict.get('task_type', None)
+    if task_type is None:
+        raise base.BaseHandler.InvalidInputException('No task_type provided')
+    target_id = task_entry_dict.get('target_id', None)
+    if target_id is None:
+        raise base.BaseHandler.InvalidInputException('No target_id provided')
+    status = task_entry_dict.get('status', None)
+    if status is None:
+        raise base.BaseHandler.InvalidInputException('No status provided')
+
+    return task_entry_dict
```

Comment:
For this and others: if you can't make a domain object out of the dict, leave a clear comment explaining why.

Reply:
Added a comment for them.

Reply:
No, this is not sufficient. You should either (a) use an existing domain object, (b) create a new domain object, or (c) explain why a domain object cannot be created.

There is, for example, a TaskEntry domain object. Why can't you use it? The comment you have written here does not give any clear explanation for that.

Reply:
Updated.

>There is, for example, a TaskEntry domain object. Why can't you use it?

The TaskEntry class requires some additional information like exploration_id, user_id, etc. Thus I think it is appropriate to create object in the handler when we have all the data in a single place.
Thank you.